File: frontend/src/app/models/apartment.dto.ts
import { ApartmentComplexDTO } from "./apartment-complex.dto";

export interface ApartmentDTO {
  id: number;
  apartmentName: string;
  apartmentLocation: string;
  numOfRooms: number;
  squareFootage: number;
  dateBuilt: string;
  currentTenantName?: string;
  currentTenantEmail?: string;
  currentTenantPhone?: string;
  apartmentComplexId: number;
}

File: backend/src/main/java/backend/DTOs/ApartmentDTO.java
package backend.DTOs;

import java.time.LocalDate;

import lombok.AllArgsConstructor;
import lombok.Data;

@AllArgsConstructor
@Data
public class ApartmentDTO {
    private long id;
    private String apartmentName;
    private String apartmentLocation;
    private Integer numOfRooms;
    private Integer squareFootage;
    private LocalDate dateBuilt;
    private String currentTenantName;
    private String currentTenantEmail;
    private String currentTenantPhone;
    private long complexId;
}
